app/services/flow/base_flow.rb

```diff
@@ -40,15 +40,26 @@ def wrap_send(handler)
     end
 
     def form_response(obj, value)
-      response = form_response?(value) ? value : FormResponse.new(success: true, errors: {})
+      response = acceptable_response_object?(value) ? value : create_form_response(value)
       obj.mark_step_complete if response.success?
       response
     end
 
+    def create_form_response(obj)
+      success = obj.respond_to?(:success?) ? obj.success? : true
+      errors = obj.respond_to?(:errors?) ? obj.errors? : {}
+      errors = {} if errors.blank?
+      FormResponse.new(success: success, errors: errors)
+    end
+
     def form_response?(response)
       response.is_a?(FormResponse) || response.is_a?(DocAuth::Response)
     end
 
+    def acceptable_response_object?(obj)
+      obj.is_a?(FormResponse) || obj.is_a?(DocAuthClient::Response)
```

Comment:
On my copy of `master`, `DocAuthClient` doesn't exist anymore... should we just replace this method with `form_response?` above?